src/sentry/runner/commands/devservices.py

```diff
@@ -288,6 +288,33 @@ def up(
                     me = os.getpid()
                     os.kill(me, signal.SIGTERM)
 
+    # Check health of services. Seperate from _start_services
+    # in case there are dependencies needed for the health
+    # check (for example: kafka's healthcheck requires zookeeper)
+    with ThreadPoolExecutor(max_workers=len(selected_services)) as executor:
+        futures = []
+        for name in selected_services:
+            futures.append(
+                executor.submit(
+                    check_health,
+                    name,
+                    containers,
+                )
+            )
+        for future in as_completed(futures):
+            # If there was an exception, reraising it here to the main thread
+            # will not terminate the whole python process. We'd like to report
+            # on this exception and stop as fast as possible, so terminate
+            # ourselves. I believe (without verification) that the OS is now
+            # free to cleanup these threads, but not sure if they'll remain running
+            # in the background. What matters most is that we regain control
+            # of the terminal.
```

Comment:
I don't think they would still remain running in the background since we kill them, but I could be wrong. My understanding is that os.kill should send a SIGTERM signal to the current process, which should take care of all threads and child processes.

Reply:
ahah this is where that errant kill is -- we should get rid of this it doesn't actually work the way the original author intended and makes debugging this impossible

Reply:
Removed the os.kill(). I'm still logging the exception. The only concern I have is the "regain control off the terminal" but I'm not sure if that is needed or how to test it. The future should resolve or throw exception, so unsure how we process futures.

Reply:
I think the comment comes from when it was using multiprocessing.pool and now that it's concurrent futures it should be fine

Reply:
SG - my read of the code is that it should be fine, but time will tell